### PR TITLE
Add CreateTeeWriter function to logutil

### PR DESCRIPTION
**Pull request by andrewsun2898**

### CODE REVIEW

**Review by 10000TB — APPROVED** [review]

File: pkg/logutil/logutil.go [hunk 1]
```diff
@@ -32,23 +33,30 @@ func NewWriterToFile(maxSize int, path string) (io.Writer, error) {
...
+}
+
+// CreateTeeWriter tees out output to a file path specified by logPath up to a max limit. Creates necessary directories for the specified logpath if they don't exist.
+func CreateTeeWriter(writer io.Writer, logPath string, maxSize int) (io.Writer, error) {
+	if logPath == "" {
+		return nil, fmt.Errorf("Empty log path")
+	}
+	dir := filepath.Dir(logPath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		log.Printf("Log directory %s doesn't exist, creating...", dir)
```
Comment (rminnich) [inline pkg/logutil/logutil.go:52]: it would be best not to have this log.Printf in a package, can it be removed?

Comment (andrewsun2898) [reply pkg/logutil/logutil.go]: Done.

File: pkg/logutil/logutil.go [hunk 2]
```diff
@@ -32,23 +33,29 @@ func NewWriterToFile(maxSize int, path string) (io.Writer, error) {
...
+	return writer, err
+}
+
+// CreateTeeWriter tees out output to a file path specified by logPath up to a max limit. Creates necessary directories for the specified logpath if they don't exist.
+func CreateTeeWriter(writer io.Writer, logPath string, maxSize int) (io.Writer, error) {
+	if logPath == "" {
+		return nil, fmt.Errorf("Empty log path")
+	}
+	dir := filepath.Dir(logPath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
```
Comment (rminnich) [inline pkg/logutil/logutil.go:51]: I just realized there is a gotcha here. You're checking to see it exists, but not checking the file type. 
Consider the case that the state succeeds, but the mode on the directory is wrong, e.g. it's a file. os.MkdirAll actually does that test. Further, the test you are doing here is done in os.MkDirAll anyway. 
I'm sorry I did not catch this, but i'd suggest skipping the stat entirely, and do
if err := os.MkdirAll(...
so remove 51, just have lines 52-54

**Review by rminnich — COMMENTED** [review]
Sorry, I realized it needed on last bit of fixing.

File: pkg/logutil/logutil.go (hunk 2 above)
Comment (andrewsun2898) [reply pkg/logutil/logutil.go]: Thanks for catching that! Done.

**Review by rminnich — APPROVED** [review]

**Pull request merged**